src/main/java/com/parsing/api/ExtractorLotId.java

```diff
@@ -0,0 +1,91 @@
+package com.parsing.api;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.parsing.model.LotId;
+import com.parsing.repository.LotIdRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.client.RestTemplate;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.time.ZonedDateTime;
+import java.util.Optional;
+
+@Slf4j
+@RequiredArgsConstructor
+@Component
+public class ExtractorLotId {
+
+    private final ObjectMapper objectMapper;
+
+    private final LotIdRepository lotIdRepository;
+
+    private final RestTemplate restTemplate;
+
+    private String offset = null;
+
+    @Value("${time.offset}")
+    private void setTimeOffset(String timeOffset) {
+        if (timeOffset != null && !timeOffset.isEmpty()) {
+            offset = timeOffset;
+        }
+    }
+
+    public void tryExtractLots() {
+        String START_DATE_URL = "https://public.api.openprocurement.org/api/2.5/tenders?offset=" + offset;
+        JsonNode jsonNode;
+        ResponseEntity<String> response;
+        URI uri;
+        try {
+            uri = new URI(START_DATE_URL);
+            log.info("URI {} saved to db", uri);
+            response = restTemplate.getForEntity(uri, String.class);
+            jsonNode = objectMapper.readTree(response.getBody());
+        } catch (URISyntaxException | JsonProcessingException e) {
+            throw new RuntimeException(e);
```

Comment:
it'd be better to create new exception class or at least to add some message regarding the issue that happened: public RuntimeException(String message, Throwable cause) - this constructor